Description: cups-browsed: SECURITY FIX: Fixed a bug in the remove_bad_chars()
 failing to reliably filter out illegal characters if there are two
 or more subsequent illegal characters, allowing execution of
 arbitrary commands with the rights of the "lp" user, using forged
 print service announcements on DNS-SD servers (Bug #1265).
Author: Till Kamppeter <till.kamppeter@gmail.com>
Last-Update: 2015-02-27
Origin: upstream
Bug-CVE: CVE-2015-2265
Bug-Debian: https://bugs.debian.org/780267

--- a/utils/cups-browsed.c
+++ b/utils/cups-browsed.c
@@ -634,6 +634,7 @@
 		       str[i] == '.' || str[i] == ','))) {
       /* Letter or number, keep it */
       havedash = 0;
+      str[j] = str[i];
     } else {
       /* Replace all other characters by a single '-' */
       if (havedash == 1)
